### PR TITLE
[UF-408]: Pull Request feature, Merge, Diff capabilities addded

### PR DESCRIPTION
**Pull request by adrielparedes**
* Merge command added based on Cherry Pick command. It's implemented on JGitFileSystemProvider.copy method.
* Diff between branches command added. Is implemented in JGitFileSystem.readAttributes method

It provides these low level methods to be used into Pull Request Service in Guvnor.

### CODE REVIEW

File: uberfire-nio2-backport/uberfire-nio2-impls/uberfire-nio2-jgit/src/main/java/org/uberfire/java/nio/fs/jgit/JGitHiddenAttributeView.java [hunk 1]
```diff
@@ -0,0 +1,29 @@
+package org.uberfire.java.nio.fs.jgit;
```
Comment (porcelli) [inline uberfire-nio2-backport/uberfire-nio2-impls/uberfire-nio2-jgit/src/main/java/org/uberfire/java/nio/fs/jgit/JGitHiddenAttributeView.java]: license

File: uberfire-nio2-backport/uberfire-nio2-impls/uberfire-nio2-jgit/src/main/java/org/uberfire/java/nio/fs/jgit/JGitHiddenAttributeView.java [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+
+import org.uberfire.java.nio.IOException;
+import org.uberfire.java.nio.base.attributes.HiddenAttributeView;
+import org.uberfire.java.nio.base.attributes.HiddenAttributes;
+import org.uberfire.java.nio.file.attribute.BasicFileAttributeView;
+import org.uberfire.java.nio.fs.jgit.util.JGitUtil;
+
+public class JGitHiddenAttributeView extends HiddenAttributeView<JGitPathImpl> {
```
Comment (porcelli) [inline uberfire-nio2-backport/uberfire-nio2-impls/uberfire-nio2-jgit/src/main/java/org/uberfire/java/nio/fs/jgit/JGitHiddenAttributeView.java]: javadoc?

File: uberfire-nio2-backport/uberfire-nio2-impls/uberfire-nio2-jgit/src/main/java/org/uberfire/java/nio/fs/jgit/daemon/filters/HiddenBranchRefFilter.java [hunk 3]
```diff
@@ -0,0 +1,26 @@
+package org.uberfire.java.nio.fs.jgit.daemon.filters;
+
+import java.util.Map;
+import java.util.regex.Pattern;
+import java.util.stream.Collectors;
+
+import org.eclipse.jgit.lib.Ref;
+import org.eclipse.jgit.transport.RefFilter;
+
+public class HiddenBranchRefFilter implements RefFilter {
```
Comment (porcelli) [inline uberfire-nio2-backport/uberfire-nio2-impls/uberfire-nio2-jgit/src/main/java/org/uberfire/java/nio/fs/jgit/daemon/filters/HiddenBranchRefFilter.java]: javadoc?

File: uberfire-nio2-backport/uberfire-nio2-impls/uberfire-nio2-jgit/src/main/java/org/uberfire/java/nio/fs/jgit/util/commands/DiffBranches.java [hunk 4]
```diff
@@ -0,0 +1,113 @@
...
+import org.eclipse.jgit.lib.ObjectLoader;
+import org.eclipse.jgit.lib.Repository;
+import org.eclipse.jgit.patch.FileHeader;
+import org.uberfire.java.nio.base.FileDiff;
+import org.uberfire.java.nio.fs.jgit.util.JGitUtil;
+import org.uberfire.java.nio.fs.jgit.util.exceptions.GitException;
+
+import static org.uberfire.commons.validation.PortablePreconditions.*;
+
+public class DiffBranches extends GitCommand {
```
Comment (porcelli) [inline uberfire-nio2-backport/uberfire-nio2-impls/uberfire-nio2-jgit/src/main/java/org/uberfire/java/nio/fs/jgit/util/commands/DiffBranches.java]: javadoc?

File: uberfire-nio2-backport/uberfire-nio2-impls/uberfire-nio2-jgit/src/main/java/org/uberfire/java/nio/fs/jgit/util/commands/DiffBranches.java [hunk 5]
```diff
@@ -0,0 +1,113 @@
+package org.uberfire.java.nio.fs.jgit.util.commands;
```
Comment (porcelli) [inline uberfire-nio2-backport/uberfire-nio2-impls/uberfire-nio2-jgit/src/main/java/org/uberfire/java/nio/fs/jgit/util/commands/DiffBranches.java]: license

File: uberfire-nio2-backport/uberfire-nio2-impls/uberfire-nio2-jgit/src/main/java/org/uberfire/java/nio/fs/jgit/util/commands/Merge.java [hunk 6]
```diff
@@ -0,0 +1,66 @@
+package org.uberfire.java.nio.fs.jgit.util.commands;
```
Comment (porcelli) [inline uberfire-nio2-backport/uberfire-nio2-impls/uberfire-nio2-jgit/src/main/java/org/uberfire/java/nio/fs/jgit/util/commands/Merge.java]: license

File: uberfire-nio2-backport/uberfire-nio2-impls/uberfire-nio2-jgit/src/main/java/org/uberfire/java/nio/fs/jgit/util/commands/Merge.java [hunk 7]
```diff
@@ -0,0 +1,66 @@
...
+import org.eclipse.jgit.lib.Repository;
+import org.eclipse.jgit.revwalk.RevCommit;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.uberfire.java.nio.fs.jgit.util.JGitUtil;
+import org.uberfire.java.nio.fs.jgit.util.exceptions.GitException;
+
+import static org.uberfire.commons.validation.PortablePreconditions.*;
+
+public class Merge extends GitCommand {
```
Comment (porcelli) [inline uberfire-nio2-backport/uberfire-nio2-impls/uberfire-nio2-jgit/src/main/java/org/uberfire/java/nio/fs/jgit/util/commands/Merge.java]: javadoc?

File: uberfire-nio2-backport/uberfire-nio2-impls/uberfire-nio2-jgit/src/test/java/org/uberfire/java/nio/fs/jgit/JGitFileSystemProviderDiffTest.java [hunk 8]
```diff
@@ -0,0 +1,110 @@
...
+        final Path path = provider.getPath( newRepo );
+        final List<FileDiff> diffs = (List<FileDiff>) provider.readAttributes( path, "diff:master,develop" ).get( "diff" );
+
+        diffs.forEach( elem -> logger.info( elem.toString() ) );
+
+        assertThat( diffs.size() ).isEqualTo( 0 );
+
+    }
+
+}
```
Comment (porcelli) [inline uberfire-nio2-backport/uberfire-nio2-impls/uberfire-nio2-jgit/src/test/java/org/uberfire/java/nio/fs/jgit/JGitFileSystemProviderDiffTest.java]: missing binary files (ie. image) to check behavior

File: uberfire-nio2-backport/uberfire-nio2-impls/uberfire-nio2-jgit/src/test/java/org/uberfire/java/nio/fs/jgit/JGitFileSystemProviderMergeTest.java [hunk 9]
```diff
@@ -0,0 +1,105 @@
...
+        }
+
+        final Path master = provider.getPath( URI.create( "git://master@merge-test-repo" ) );
+        final Path develop = provider.getPath( URI.create( "git://develop@merge-test-repo" ) );
+
+        provider.copy( develop, null, new MergeCopyOption() );
+
+    }
+
+}
```
Comment (porcelli) [inline uberfire-nio2-backport/uberfire-nio2-impls/uberfire-nio2-jgit/src/test/java/org/uberfire/java/nio/fs/jgit/JGitFileSystemProviderMergeTest.java]: binary files missing.

File: uberfire-nio2-backport/uberfire-nio2-impls/uberfire-nio2-jgit/src/test/java/org/uberfire/java/nio/fs/jgit/JGitFileSystemProviderMergeTest.java [hunk 10]
```diff
@@ -0,0 +1,105 @@
+package org.uberfire.java.nio.fs.jgit;
```
Comment (porcelli) [inline uberfire-nio2-backport/uberfire-nio2-impls/uberfire-nio2-jgit/src/test/java/org/uberfire/java/nio/fs/jgit/JGitFileSystemProviderMergeTest.java]: license

File: uberfire-nio2-backport/uberfire-nio2-impls/uberfire-nio2-jgit/src/test/java/org/uberfire/java/nio/fs/jgit/JGitFileSystemProviderDiffTest.java [hunk 11]
```diff
@@ -0,0 +1,110 @@
+package org.uberfire.java.nio.fs.jgit;
```
Comment (porcelli) [inline uberfire-nio2-backport/uberfire-nio2-impls/uberfire-nio2-jgit/src/test/java/org/uberfire/java/nio/fs/jgit/JGitFileSystemProviderDiffTest.java]: license

File: uberfire-nio2-backport/uberfire-nio2-impls/uberfire-nio2-jgit/src/test/java/org/uberfire/java/nio/fs/jgit/JGitMergeTest.java [hunk 12]
```diff
@@ -0,0 +1,130 @@
+package org.uberfire.java.nio.fs.jgit;
```
Comment (porcelli) [inline uberfire-nio2-backport/uberfire-nio2-impls/uberfire-nio2-jgit/src/test/java/org/uberfire/java/nio/fs/jgit/JGitMergeTest.java]: license

File: uberfire-nio2-backport/uberfire-nio2-impls/uberfire-nio2-jgit/src/test/java/org/uberfire/java/nio/fs/jgit/JGitMergeTest.java [hunk 13]
```diff
@@ -0,0 +1,130 @@
...
+
+        commit( origin, "develop", "name", "name@example.com", "develop-4", null, null, false, new HashMap<String, File>() {{
+            put( "file5.txt", tempFile( "temp5" ) );
+        }} );
+
+        new Merge( origin, "develop", "nonexistent" ).execute();
+
+    }
+
+}
```
Comment (porcelli) [inline uberfire-nio2-backport/uberfire-nio2-impls/uberfire-nio2-jgit/src/test/java/org/uberfire/java/nio/fs/jgit/JGitMergeTest.java]: add binary case

File: uberfire-nio2-backport/uberfire-nio2-impls/uberfire-nio2-jgit/src/test/java/org/uberfire/java/nio/fs/jgit/daemon/filters/HiddenBranchRefFilterTest.java [hunk 14]
```diff
@@ -0,0 +1,47 @@
+package org.uberfire.java.nio.fs.jgit.daemon.filters;
```
Comment (porcelli) [inline uberfire-nio2-backport/uberfire-nio2-impls/uberfire-nio2-jgit/src/test/java/org/uberfire/java/nio/fs/jgit/daemon/filters/HiddenBranchRefFilterTest.java]: license

File: uberfire-nio2-backport/uberfire-nio2-model/src/main/java/org/uberfire/java/nio/base/FileDiff.java [hunk 15]
```diff
@@ -0,0 +1,102 @@
+package org.uberfire.java.nio.base;
+
+import java.util.List;
+
+import static org.uberfire.commons.validation.PortablePreconditions.*;
+
+public class FileDiff {
```
Comment (porcelli) [inline uberfire-nio2-backport/uberfire-nio2-model/src/main/java/org/uberfire/java/nio/base/FileDiff.java]: javadoc

File: uberfire-nio2-backport/uberfire-nio2-model/src/main/java/org/uberfire/java/nio/base/FileDiff.java [hunk 16]
```diff
@@ -0,0 +1,102 @@
+package org.uberfire.java.nio.base;
```
Comment (porcelli) [inline uberfire-nio2-backport/uberfire-nio2-model/src/main/java/org/uberfire/java/nio/base/FileDiff.java]: license

File: uberfire-nio2-backport/uberfire-nio2-model/src/main/java/org/uberfire/java/nio/base/FileDiff.java [hunk 17]
```diff
@@ -0,0 +1,102 @@
...
+        builder.append( this.getNameB() );
+        builder.append( " -> " );
+        builder.append( "( " + this.getStartB() + " , " + this.getEndB() + " )" );
+        builder.append( "[ " + linesFromB + " ]" );
+        builder.append( "}" );
+
+        return builder.toString();
+    }
+
+}
```
Comment (porcelli) [inline uberfire-nio2-backport/uberfire-nio2-model/src/main/java/org/uberfire/java/nio/base/FileDiff.java]: equals and hashcode

File: uberfire-nio2-backport/uberfire-nio2-model/src/main/java/org/uberfire/java/nio/base/attributes/HiddenAttributeView.java [hunk 18]
```diff
@@ -0,0 +1,46 @@
+package org.uberfire.java.nio.base.attributes;
```
Comment (porcelli) [inline uberfire-nio2-backport/uberfire-nio2-model/src/main/java/org/uberfire/java/nio/base/attributes/HiddenAttributeView.java]: license

File: uberfire-nio2-backport/uberfire-nio2-model/src/main/java/org/uberfire/java/nio/base/attributes/HiddenAttributeView.java [hunk 19]
```diff
@@ -0,0 +1,46 @@
...
+import java.util.HashMap;
+import java.util.Map;
+
+import org.uberfire.java.nio.IOException;
+import org.uberfire.java.nio.base.AbstractBasicFileAttributeView;
+import org.uberfire.java.nio.file.Path;
+
+import static org.uberfire.commons.validation.PortablePreconditions.*;
+
+public abstract class HiddenAttributeView<P extends Path>
```
Comment (porcelli) [inline uberfire-nio2-backport/uberfire-nio2-model/src/main/java/org/uberfire/java/nio/base/attributes/HiddenAttributeView.java]: javadoc

File: uberfire-nio2-backport/uberfire-nio2-model/src/main/java/org/uberfire/java/nio/base/options/MergeCopyOption.java [hunk 20]
```diff
@@ -0,0 +1,7 @@
+package org.uberfire.java.nio.base.options;
```
Comment (porcelli) [inline uberfire-nio2-backport/uberfire-nio2-model/src/main/java/org/uberfire/java/nio/base/options/MergeCopyOption.java]: license

File: uberfire-nio2-backport/uberfire-nio2-model/src/main/java/org/uberfire/java/nio/base/options/MergeCopyOption.java [hunk 21]
```diff
@@ -0,0 +1,7 @@
...
+
+import org.uberfire.java.nio.file.CopyOption;
+
+public class MergeCopyOption implements CopyOption {
```
Comment (porcelli) [inline uberfire-nio2-backport/uberfire-nio2-model/src/main/java/org/uberfire/java/nio/base/options/MergeCopyOption.java]: javadoc?

**Comment by adrielparedes** [comment]
[PullRequestService -> License](https://trello.com/c/bmu3Lg5p/2-pullrequestservice-license)

File: uberfire-nio2-backport/uberfire-nio2-impls/uberfire-nio2-jgit/src/main/java/org/uberfire/java/nio/fs/jgit/util/JGitUtil.java [hunk 22]
```diff
@@ -1310,6 +1316,87 @@ public static RevCommit getLastCommit( final Git git,
...
+        }
+    }
+
+    public static HiddenAttributes buildHiddenAttributes( final JGitFileSystem fileSystem,
+                                                          final String branchName,
+                                                          final String path ) {
+
+        final BasicFileAttributes attributes = buildBasicAttributes( fileSystem, branchName, path );
+
+        return new HiddenAttributes() {
```
Comment (ederign) [inline uberfire-nio2-backport/uberfire-nio2-impls/uberfire-nio2-jgit/src/main/java/org/uberfire/java/nio/fs/jgit/util/JGitUtil.java]: Please, extract that to a class file.

File: uberfire-nio2-backport/uberfire-nio2-impls/uberfire-nio2-jgit/src/test/java/org/uberfire/java/nio/fs/jgit/JGitFileSystemProviderDiffTest.java (hunk 8 above)
Comment (ederign) [inline uberfire-nio2-backport/uberfire-nio2-impls/uberfire-nio2-jgit/src/test/java/org/uberfire/java/nio/fs/jgit/JGitFileSystemProviderDiffTest.java]: +1

File: uberfire-nio2-backport/uberfire-nio2-model/src/main/java/org/uberfire/java/nio/base/options/MergeCopyOption.java [hunk 23]
```diff
@@ -0,0 +1,28 @@
...
+package org.uberfire.java.nio.base.options;
+
+import org.uberfire.java.nio.file.CopyOption;
+
+/**
+ * This is the CopyOption that allows to merge two branches
+ * when executing copy method.
+ * You have to apply it as the third parameter of FileSystemProvider.copy() method.
+ */
+public class MergeCopyOption implements CopyOption {
```
Comment (ederign) [inline uberfire-nio2-backport/uberfire-nio2-model/src/main/java/org/uberfire/java/nio/base/options/MergeCopyOption.java]: This is just a marker? Why not a Enum?

File: uberfire-nio2-backport/uberfire-nio2-impls/uberfire-nio2-jgit/src/main/java/org/uberfire/java/nio/fs/jgit/util/commands/Merge.java (hunk 6 above)
Comment (porcelli) [inline uberfire-nio2-backport/uberfire-nio2-impls/uberfire-nio2-jgit/src/main/java/org/uberfire/java/nio/fs/jgit/util/commands/Merge.java]: @adrielparedes still missing license

File: uberfire-nio2-backport/uberfire-nio2-impls/uberfire-nio2-jgit/src/main/java/org/uberfire/java/nio/fs/jgit/util/commands/Merge.java [hunk 24]
```diff
@@ -0,0 +1,97 @@
...
+                  final String sourceBranch,
+                  final String targetBranch ) {
+
+        this.git = checkNotNull( "git", git );
+        this.sourceBranch = checkNotEmpty( "sourceBranch", sourceBranch );
+        this.targetBranch = checkNotEmpty( "targetBranch", targetBranch );
+    }
+
+    @Override
+    public Optional<List<String>> execute() {
```
Comment (porcelli) [inline uberfire-nio2-backport/uberfire-nio2-impls/uberfire-nio2-jgit/src/main/java/org/uberfire/java/nio/fs/jgit/util/commands/Merge.java]: optional of list?! just list doesn't sufice?

Comment (porcelli) [inline uberfire-nio2-backport/uberfire-nio2-impls/uberfire-nio2-jgit/src/main/java/org/uberfire/java/nio/fs/jgit/util/commands/Merge.java]: k, i see this is related to gitCommand intreface

File: uberfire-nio2-backport/uberfire-nio2-impls/uberfire-nio2-jgit/src/test/java/org/uberfire/java/nio/fs/jgit/JGitFileSystemProviderDiffTest.java (hunk 11 above)
Comment (porcelli) [inline uberfire-nio2-backport/uberfire-nio2-impls/uberfire-nio2-jgit/src/test/java/org/uberfire/java/nio/fs/jgit/JGitFileSystemProviderDiffTest.java]: ops... 2 licenses?! ;)

File: uberfire-nio2-backport/uberfire-nio2-impls/uberfire-nio2-jgit/src/main/java/org/uberfire/java/nio/fs/jgit/daemon/filters/HiddenBranchRefFilter.java [hunk 25]
```diff
@@ -0,0 +1,52 @@
...
+
+    }
+
+    /**
+     * Checks if a branch name matches the hidden branch regexp
+     * @param branch the branch you want to check.
+     * @return return if the branch is hidden or not
+     */
+    public static boolean isHidden( String branch ) {
+        return Pattern.matches( HIDDEN_BRANCH_REGEXP, branch );
```
Comment (porcelli) [inline uberfire-nio2-backport/uberfire-nio2-impls/uberfire-nio2-jgit/src/main/java/org/uberfire/java/nio/fs/jgit/daemon/filters/HiddenBranchRefFilter.java]: better compile it and use a compiled version of it

File: uberfire-nio2-backport/uberfire-nio2-model/src/main/java/org/uberfire/java/nio/base/attributes/HiddenAttributes.java [hunk 26]
```diff
@@ -0,0 +1,25 @@
...
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.uberfire.java.nio.base.attributes;
+
+import org.uberfire.java.nio.file.attribute.BasicFileAttributes;
+
+public interface HiddenAttributes extends BasicFileAttributes {
```
Comment (porcelli) [inline uberfire-nio2-backport/uberfire-nio2-model/src/main/java/org/uberfire/java/nio/base/attributes/HiddenAttributes.java]: javadoc?

File: uberfire-nio2-backport/uberfire-nio2-model/src/main/java/org/uberfire/java/nio/base/attributes/HiddenAttributesImpl.java [hunk 27]
```diff
@@ -0,0 +1,83 @@
...
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.uberfire.java.nio.base.attributes;
+
+import org.uberfire.java.nio.file.attribute.BasicFileAttributes;
+import org.uberfire.java.nio.file.attribute.FileTime;
+
+public class HiddenAttributesImpl implements HiddenAttributes {
```
Comment (porcelli) [inline uberfire-nio2-backport/uberfire-nio2-model/src/main/java/org/uberfire/java/nio/base/attributes/HiddenAttributesImpl.java]: javadoc?

File: uberfire-nio2-backport/uberfire-nio2-impls/uberfire-nio2-jgit/src/main/java/org/uberfire/java/nio/fs/jgit/util/commands/Merge.java (hunk 24 above)
Comment (ederign) [inline uberfire-nio2-backport/uberfire-nio2-impls/uberfire-nio2-jgit/src/main/java/org/uberfire/java/nio/fs/jgit/util/commands/Merge.java]: @porcelli that is also my concern at first. I don't see in that code the case of having a empty/null list. but maybe I'm missing something,

File: uberfire-nio2-backport/uberfire-nio2-model/src/main/java/org/uberfire/java/nio/base/attributes/HiddenAttributeView.java [hunk 28]
```diff
@@ -0,0 +1,67 @@
...
+
+    public abstract HiddenAttributes readAttributes() throws IOException;
+
+    @Override
+    public Map<String, Object> readAttributes( final String... attributes ) {
+        final HiddenAttributes attrs = readAttributes();
+
+        return new HashMap<String, Object>( super.readAttributes( attributes ) ) {{
+            for ( final String attribute : attributes ) {
+                checkNotEmpty( "attribute", attribute );
```
Comment (ederign) [inline uberfire-nio2-backport/uberfire-nio2-model/src/main/java/org/uberfire/java/nio/base/attributes/HiddenAttributeView.java]: what this line does? a break if attribute is empty?

File: uberfire-nio2-backport/uberfire-nio2-model/src/main/java/org/uberfire/java/nio/base/attributes/HiddenAttributeView.java [hunk 29]
```diff
@@ -0,0 +1,67 @@
...
+
+/**
+ * This is a view that extends a Basic Attribute View and adds the "isHidden" attribute.
+ * That attribute lets you know if the branch you are querying is a hidden branch or not.
+ * Hidden branches should not be used, are just a mechanism to merge.
+ */
+public abstract class HiddenAttributeView<P extends Path>
+        extends AbstractBasicFileAttributeView<P> {
+
+    public static final String VERSION = "hidden";
```
Comment (porcelli) [inline uberfire-nio2-backport/uberfire-nio2-model/src/main/java/org/uberfire/java/nio/base/attributes/HiddenAttributeView.java]: why a field named VERSION has a valur of "hidden"?

File: uberfire-nio2-backport/uberfire-nio2-impls/uberfire-nio2-jgit/src/main/java/org/uberfire/java/nio/fs/jgit/util/commands/Merge.java [hunk 30]
```diff
@@ -0,0 +1,113 @@
...
+    private void checkCanCommit( final Repository repo,
+                                 final RevCommit commonAncestor,
+                                 final RevCommit sourceCommitTree,
+                                 final RevCommit targetCommitTree,
+                                 final String sourceBranch,
+                                 final String targetBranch ) {
+        try {
+            ThreeWayMerger merger = MergeStrategy.RECURSIVE.newMerger( repo, true );
+            merger.setBase( commonAncestor );
+            boolean canMerge = merger.merge( sourceCommitTree, targetCommitTree );
```
Comment (porcelli) [inline uberfire-nio2-backport/uberfire-nio2-impls/uberfire-nio2-jgit/src/main/java/org/uberfire/java/nio/fs/jgit/util/commands/Merge.java]: didMerge better name?

File: uberfire-nio2-backport/uberfire-nio2-model/src/main/java/org/uberfire/java/nio/base/attributes/HiddenAttributeView.java [hunk 31]
```diff
@@ -0,0 +1,64 @@
...
+
+        return new HashMap<String, Object>( super.readAttributes( attributes ) ) {{
+
+            for ( final String attribute : attributes ) {
+                checkNotEmpty( "attribute", attribute );
+
+                if ( attribute.equals( "*" ) || attribute.equals( HIDDEN ) ) {
+                    put( HIDDEN, attrs.isHidden() );
+                }
+            }
```
Comment (porcelli) [inline uberfire-nio2-backport/uberfire-nio2-model/src/main/java/org/uberfire/java/nio/base/attributes/HiddenAttributeView.java]: where's the break for "attribute.equals( "*" ) " ? that's a good optimization used elsewhere. please add it back

**Comment by ederign** [comment]
Thanks @adrielparedes

**Pull request merged**